File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/example/thymeleafjava/web/TestController.java
package de.tschuehly.example.thymeleafjava.web;

import de.tschuehly.example.thymeleafjava.web.index.IndexViewComponent;
import de.tschuehly.example.thymeleafjava.web.layout.LayoutViewComponent;
import de.tschuehly.example.thymeleafjava.web.simple.SimpleViewComponent;
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class TestController {

  private final SimpleViewComponent simpleViewComponent;
  private final IndexViewComponent indexViewComponent;
  private final LayoutViewComponent layoutViewComponent;


  public TestController(
      SimpleViewComponent simpleViewComponent,
      IndexViewComponent indexViewComponent,
      LayoutViewComponent layoutViewComponent
  ) {
    this.simpleViewComponent = simpleViewComponent;
    this.indexViewComponent = indexViewComponent;
    this.layoutViewComponent = layoutViewComponent;
  }

  @GetMapping("/")
  ViewContext indexComponent() {
    return indexViewComponent.render();
  }

  @GetMapping("/simple")
  ViewContext simple() {
    return simpleViewComponent.render();
  }

  @GetMapping("/layout")
  ViewContext layoutComponent() {
    return layoutViewComponent.render(simpleViewComponent.render());
  }

  @GetMapping("/resource-template")
  String templateTest() {
    return "template-test";
  }
}

File: examples/thymeleaf-kotlin-example/src/main/kotlin/de/tschuehly/example/thymeleafkotlin/web/TestController.kt
package de.tschuehly.example.thymeleafkotlin.web

import de.tschuehly.example.thymeleafkotlin.web.index.IndexViewComponent
import de.tschuehly.example.thymeleafkotlin.web.layout.LayoutViewComponent
import de.tschuehly.example.thymeleafkotlin.web.simple.SimpleViewComponent
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

@Controller
class TestController(
    private val simpleViewComponent: SimpleViewComponent,
    private val indexViewComponent: IndexViewComponent,
    private val layoutViewComponent: LayoutViewComponent
) {

    @GetMapping("/")
    fun indexComponent() = indexViewComponent.render()

    @GetMapping("/simple")
    fun simpleComponent() = simpleViewComponent.render()

    @GetMapping("/layout")
    fun layoutComponent() = layoutViewComponent.render(simpleViewComponent.render())


    @GetMapping("/resource-template")
    fun templateTest(): String {
        return "template-test"
    }
}